source/binary.cpp

```diff
@@ -45,6 +45,21 @@ spv_result_t spvBinaryHeaderGet(const spv_const_binary binary,
   // TODO: Validation checking?
   pHeader->magic = spvFixWord(binary->code[SPV_INDEX_MAGIC_NUMBER], endian);
   pHeader->version = spvFixWord(binary->code[SPV_INDEX_VERSION_NUMBER], endian);
+  // Assumes versions are composed of a major and minor number, where major = 1
+  // and minor {0, current spec minor}. If version 2.x or 1.1x is released this
+  // check will need to be updated.
+
+  bool is_valid_version = false;
+  for (uint32_t minor = 0;
```

Comment:
Why not just?
```
// Minimum version was 1.0 and max version is defined by SPV_VERSION.
if (pHeader->version < SPV_SPIRV_VERSION_WORD(1, 0) || pHeader->version > SPV_VERSION)
  return SPV_ERROR_INVALID_BINARY;
```

This should handle all future versions too.

Reply:
That is a much more elegant solution.